src/module.c

```diff
@@ -6507,6 +6507,49 @@ const char *RM_CallReplyProto(RedisModuleCallReply *reply, size_t *len) {
     return callReplyGetProto(reply, len);
 }
 
+/* Redis only exposes "side effects" of command execution at the end of an execution unit.
+ *
+ * Side effects currently handled are
+ * - propagating replication/AOF operations
+ * - executing the commands of clients blocked on keys modified by commands
+ *
+ * In general, for modules, execution unit start and end with the taking/release of the GIL.  For module code executed
+ * on the Redis main thread, this is done implicitly by Redis itself, while for module code executed on a separate
+ * thread, this is done explicitly by the module itself.
+ *
+ * This function enables modules to flush the execution unit within the middle of an execution unit, thereby having
+ * side effects become visible and impact future operations that occur within the execution unit.
+ *
+ * Usage Example: a module runs multiple commands via RM_Call within a single execution unit, but wants
+ * processCommand() like semantics, where blocked clients that could be unblocked by a previous RM_Call operation will
+ * be executed before the next RM_Call is made.  By calling RM_FlushExecutionUnit between RM_Call operations, the
+ * module will get those semantics.
+ *
+ * The flags is a bit mask of these:
+ *
+ * - `REDISMODULE_FLUSH_EXEC_UNIT_FLAG_DEFAULT`: We propagate pending command operations (AOF/Replication) and unblock
+ *                                               clients
+ */
+int RM_FlushExecutionUnit(RedisModuleCtx *ctx, unsigned int flags) {
```

Comment:
```suggestion
int RM_FlushPendingCommands(RedisModuleCtx *ctx, unsigned int flags) {
```

Reply:
i'm certainly ok with the term "Pending" instead of "ExecutionUnit", but i think that "commands" could be confusing.
someone can think that this generates commands, where in fact it's the side effects of the commands already executed.
maybe "FlushPendingEffects"?

Reply:
I could live with FlushPendingEffects. It seems to be the right intention

Reply:
I think FlushContext is too generic, context is used for so many things.
 
CompleteTransacton seems ok to me (p.s. it calls propagatePendingCommands which will wrap what you did with multi-exec).
i'm a little bit uncomfortable with using "complete" since it might suggest that we needed to explicitly "start" it too, but FlushTransaction is not much better.. maybe there's a better word, but if not, i'm ok with `RM_CompleteTransacton`.

Reply:
I'm OK with `RM_CompleteTransaction`.

Reply:
Ok, I spend some time thinking, I can't come up with anything else.